azuread/resource_user.go

```diff
@@ -201,6 +213,11 @@ func resourceUserUpdate(d *schema.ResourceData, meta interface{}) error {
 		userUpdateParameters.UsageLocation = p.String(usageLocation)
 	}
 
+	if d.HasChange("immutable_id") {
```

Comment:
If this is immutable is it really updateable? 🤔

Reply:
It appears to be an unfortunate naming convention. From the documents it can be added to the user after creation, and it can be modified once set.